### DISCUSSION TITLE
How to build a test plan or suite for performance testing using artillery for Https APIs

### DISCUSSION THREAD

**Discussion by amank575** [artilleryio/artillery#1658] [Artillery]
Hello team, Since I was performing load testing on my company's APIs, a test suite or plan that would perform performance testing on all of our APIs and generate an HTML report was required. I am now unable to implement it because I have no idea how to bundle or create a test suite for the issue. So do we have some functionality to make a test plan or test suite with artillery?

**Comment by hassy** [comment]
hi @amank575 👋 

Please share more specifics of what you're trying to do. Typical approach is:

- Write user scenarios for the APIs and services you are going to be testing
- Create phase configurations to put load on the systems you're testing
- Run those tests - manually, on a schedule, or from a CICD pipeline. Each test can generate a report
- Share those reports with the stakeholders

**Reply by amank575** [reply]
hi @hassy My approach :

- Write different user scenarios for all  different APIs .
- Automate load testing with generation of reports of APIs in my CICD pipeline.
- Review created reports 

for reference i have few APIs 

- addUserDetail 
- editUser
- searchUser etc...
Now I want to automate the process of load testing on those APIs one after another and create report file at the end of it . Like I define test plan or test suite for it and can automate the process.
Can you point me in the direction from which I can achieve the said process .
